packages/@expo/prebuild-config/src/plugins/unversioned/expo-splash-screen/withAndroidSplashStrings.ts

```diff
@@ -4,14 +4,28 @@ import { getAndroidSplashConfig } from './getAndroidSplashConfig';
 
 const RESIZE_MODE_KEY = 'expo_splash_screen_resize_mode';
 const STATUS_BAR_TRANSLUCENT_KEY = 'expo_splash_screen_status_bar_translucent';
+const FADE_DURATION_MS_KEY = 'expo_splash_screen_fade_duration_ms';
 
-export const withAndroidSplashStrings: ConfigPlugin = (config) => {
+type ExtraProps = {
+  resizeMode?: string;
+  fadeDurationMs?: number;
+};
+
+const defaultResizeMode = 'contain';
+
+export const withAndroidSplashStrings: ConfigPlugin<ExtraProps> = (config, splash) => {
   return withStringsXml(config, (config) => {
     const splashConfig = getAndroidSplashConfig(config);
     if (splashConfig) {
-      const { resizeMode } = splashConfig;
+      const resizeMode = splash?.resizeMode || defaultResizeMode;
       const statusBarTranslucent = !!config.androidStatusBar?.translucent;
-      config.modResults = setSplashStrings(config.modResults, resizeMode, statusBarTranslucent);
+      const fadeDurationMs = `${splash?.fadeDurationMs}`;
```

Comment:
so that the `fadeDurationMs` could be an "undefined" string and set to xml. is that expected?
maybe the better way is only to add the xml element if the fadeDurationMs is not undefined.

Reply:
I've modified this to make sure the fadeDurationMs defaults to 0 if it is not defined either in the splash config or in extra props.

Reply:
Added a new validation function for the fade time, and added warnings if `app.json` contains a value outside the allowed limits (0 to 5000 ms).